app/containers/TracksContainer/TrackDetailsContainer/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ *
+ * TrackDetailsContainer Container
+ *
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { useParams } from 'react-router-dom';
+import { selectSingleTrackLoading, selectTrackDetails } from '../selectors';
+import { tracksContainerCreators } from '../reducer';
+import { TrackComponent } from '@app/components/TrackComponent/index';
+import If from '@components/If';
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 2rem 0;
+`;
+
+const TrackDetailsWrapper = styled.div`
+  padding: 2rem;
+`;
+
+export function TrackDetailsContainer({ dispatchRequestTrackDetails, trackDetails, singleTrackLoading }) {
+  const { trackId } = useParams();
+
+  useEffect(() => {
+    dispatchRequestTrackDetails(parseInt(trackId));
+  }, [trackId]);
+
+  return (
+    <Wrapper>
+      <If condition={!singleTrackLoading}>
```

Comment:
it should not be with loading. We should check if it is empty or no. If it is empty we dont even get in

Reply:
Updated accordingly.